src/test/java/com/amazon/opendistroforelasticsearch/knn/index/KNNCircuitBreakerIT.java

```diff
@@ -41,10 +41,11 @@ private void tripCb() throws Exception {
         // Set circuit breaker limit to 1 KB
         updateClusterSettings("knn.memory.circuit_breaker.limit", "1kb");
 
-        // Create Single Shard Index so that all data is hosted on a single node
+        // Create index with 1 primary and numNodes-1 replicas so that the data will be on every node in the cluster
+        int numNodes = Integer.parseInt(System.getProperty("cluster.number_of_nodes"));
         Settings settings = Settings.builder()
                 .put("number_of_shards", 1)
-                .put("number_of_replicas", 2)
+                .put("number_of_replicas", numNodes - 1)
```

Comment:
+1.

Are there any other tests with replicas that need to be taken care of?

Reply:
`numNodes - 1` and the 1 primary ensures there is one shard per node, correct?

This test fails with only one shard on a multi-node cluster because it requires a shard from 2 different indices to be searched on the same node to trigger the circuit breaker. With this fix, `./gradlew build` passes.